common/decisions/wc_cen_kal_decisions.txt

```diff
@@ -0,0 +1,370 @@
+		exists = var:depot_location
+		var:depot_location.province_owner = root
+	}
+	
+	cost = { prestige = 50 }
+	
+	effect = {
+		trigger_event = cen_kal.1006
+		custom_tooltip = change_depot_policy_tt_basic
+	}
+}
+
+blood_shards_gather_operation = {
+	ai_check_interval = 12
+	picture = "gfx/interface/illustrations/decisions/quilboar_gatherers.dds"
+	desc = blood_shards_gather_operation_desc
+	title = blood_shards_gather_operation_name
+	selection_tooltip = blood_shards_gather_operation_tt
+	confirm_text = blood_shards_gather_operation_confirm
+	major = yes
+
+	cooldown = { years = 2 }
+
+	is_shown = {
+		is_landed = yes
+		faith = faith:agamaggan_worship
+		NOT = { root.primary_title = { is_mercenary_company = yes } }
+	}
+	
+	is_valid = {
+		custom_tooltip = {
+		  text = blood_shards_gather_operation_req_tt
+		  NOT = { has_character_flag = has_working_bloodstone_foragers }
+		}
+	}
+	
+	cost = { gold = 75 piety = 50 }
+	
+	effect = { trigger_event = cen_kal.1050 }
+}
+
+change_agamaggan_blessing = {
+	ai_check_interval = 12
+	picture = "gfx/interface/illustrations/decisions/quilboar_ritual.dds"
+	desc = change_agamaggan_blessing_desc
+	title = change_agamaggan_blessing_name
+	selection_tooltip = change_agamaggan_blessing_tt
+	confirm_text = change_agamaggan_blessing_confirm
+	major = yes
+
+	cooldown = { years = 1 }
+
+	is_shown = {
+		is_landed = yes
+		faith = faith:agamaggan_worship
+		NOT = { root.primary_title = { is_mercenary_company = yes } }
+	}
+	
+	is_valid = {
+		custom_tooltip = {
+		  text = change_agamaggan_blessing_req_tt
+		  var:blood_shard_count >= 4
+		}
+	}
+	
+	cost = { piety = 25 }
+	
+	effect = { trigger_event = cen_kal.1052 }
+}
+
+create_thorns_quilboar = {
+	ai_check_interval = 12
+	picture = "gfx/interface/illustrations/decisions/quilboar_vines.dds"
+	desc = create_thorns_quilboar_desc
+	title = create_thorns_quilboar_name
+	selection_tooltip = create_thorns_quilboar_tt
+	confirm_text = create_thorns_quilboar_confirm
+	major = yes
+
+	cooldown = { years = 2 }
+
+	is_shown = {
+		is_landed = yes
+		exists = capital_province
+		faith = faith:agamaggan_worship
+		NOR = { 
+			root.primary_title = { is_mercenary_company = yes }
+			government_has_flag = government_is_theocracy
+		}
+	}
+
+	is_valid = {
+		highest_held_title_tier >= tier_duchy
+		custom_tooltip = {
+		  text = create_thorns_quilboar_req_tt
+		  var:blood_shard_count >= 12
+		}
+	}
+	
+	effect = {
+	custom_tooltip = create_thorns_quilboar_tt
+	random_sub_realm_county = {
+		limit = {
+			OR = {
+				holder = root
+				holder = { is_vassal_or_below_of = root }
+			}
+			title_province = {
+				NOR = {
+					has_building = quilboar_thorns_01
+					has_building = quilboar_thorns_02
+					has_building = quilboar_thorns_03
+				}
+			}
+		}
+		title_province = { save_scope_as = thorn_location }
+	}
+	#Found one
+	if = { limit = { exists = scope:thorn_location } 
+		set_variable = {
+			name = thorn_location
+			value = scope:thorn_location
+		}
```

Comment:
```suggestion
```
Why do you need variable? Just use scope in localization instead of `ROOT.Var('thorn_location')`

Reply:
It is needed because the later events are not chained with this event so it needs to be saved for reference